### PR TITLE
Assert all the things!

### LINKED ISSUE #507: Double check that we're asserting in all of the right places
Every type-specific function should assert the resource is actually of that type. There are probably other places we can assert, too.

### PR DESCRIPTION
**Pull request by emersion**
Fixes #507

### CODE REVIEW

File: backend/drm/drm.c [hunk 1]
```diff
@@ -205,16 +205,22 @@ void finish_drm_resources(struct wlr_drm_backend *drm) {
 	free(drm->planes);
 }
 
+static struct wlr_drm_connector *get_drom_connector_from_output(
```
Comment (SirCmpwn) [inline backend/drm/drm.c]: >drom

File: backend/libinput/events.c [hunk 2]
```diff
@@ -150,10 +163,12 @@ static void handle_device_added(struct wlr_libinput_backend *backend,
...
 		wlr_signal_emit_safe(&backend->backend.events.new_input, wlr_dev);
 	}
-	if (libinput_device_has_capability(libinput_dev, LIBINPUT_DEVICE_CAP_GESTURE)) {
+	if (libinput_device_has_capability(libinput_dev,
+			LIBINPUT_DEVICE_CAP_GESTURE)) {
 		// TODO
 	}
-	if (libinput_device_has_capability(libinput_dev, LIBINPUT_DEVICE_CAP_SWITCH)) {
+	if (libinput_device_has_capability(libinput_dev,
+			LIBINPUT_DEVICE_CAP_SWITCH)) {
```
Comment (SirCmpwn) [inline backend/libinput/events.c]: I think these might be better balanced if you cut after the `(`

File: backend/headless/backend.c [hunk 3]
```diff
@@ -7,10 +7,17 @@
 #include <wlr/util/log.h>
 #include "backend/headless.h"
 #include "glapi.h"
+#include "util/signal.h"
+
+struct wlr_headless_backend *headless_backend_from_backend(
+		struct wlr_backend *wlr_backend) {
```
Comment (agx) [inline backend/headless/backend.c]: Should this be `get_headless_backend_from_backend()` to be consistent with the other `get_foo_from_bar()`'s above?

Comment (emersion) [reply backend/headless/backend.c]: The others have a `get_` prefix to prevent clashes with library names (`drm`, `libinput`, `wl`, etc...).

**Comment by emersion** [comment]
Addressed @SirCmpwn's comments

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**